### ISSUE TITLE
Context Attribute Do Not Support dash in V1.0 Spec

### ISSUE THREAD

**Issue by boblee0717** [cloudevents/spec#753]
In v1.0 spec, `CloudEvents attribute names MUST consist of lower-case letters ('a' to 'z') or digits ('0' to '9') from the ASCII character set.` which make the upgrading is incompatible.

**Comment by boblee0717** [comment]
I think ASCII characters like dash or underscore should be allowable. Because HTTP headers also allow these characters.

**Comment by boblee0717** [comment]
If you want to transform HTTP headers as Event Extensions, the name could be invalid according to the V1.0 spec.

**Comment by boblee0717** [comment]
I also checked the code from the sdk-go, only checked the extension name in V2 SDK for spec v1.0. No check for V0.2 V0.3 in V1 SDK. There is no check for V1.0 spec in V1 SDK.

**Comment by duglin** [comment]
what do you mean by "which make the upgrading is incompatible" ?

**Comment by clemensv** [comment]
https://github.com/cloudevents/spec/issues/265
https://github.com/cloudevents/spec/issues/367

There have been very lengthy discussions about this, which can be found in the closed issues. The two above are mere starting points. For a combination of reasons, we landed on the convention that requires lower-case letters without permitting separators.

Some programming languages have identifiers with '-', some require mapping to '_'. In some protocols, special characters may not be allowed for headers. That all is a substantial source for bugs and confusion for wire-interop and therefore we landed where we are.

This is by design.
